clients/da-vinci-client/src/main/java/com/linkedin/davinci/listener/response/AdminResponse.java

```diff
@@ -65,7 +67,19 @@ public void addStoreVersionState(StoreVersionState storeVersionState) {
     try (ByteArrayOutputStream output = new ByteArrayOutputStream()) {
       GenericDatumWriter<Object> avroDatumWriter = new GenericDatumWriter<>(StoreVersionState.SCHEMA$);
       Encoder byteToHexJsonEncoder = new ByteBufferToHexFormatJsonEncoder(StoreVersionState.SCHEMA$, output);
-      avroDatumWriter.write(storeVersionState, byteToHexJsonEncoder);
+      StoreVersionState updatedState = storeVersionState;
```

Comment:
What is the intent of this? Creating a new reference to the same object on the heap? Modifying `updatedState` will also modify `storeVersionState`, is it intentional? If you do not wish to modify the input param, then we would need to do a deep copy. Later on in the code you are seemingly doing a deep copy, but it ends up being a copy onto itself. Maybe this line should be `StoreVersionState updatedState = new StoreVersionState();` instead? Or alternatively, something like this:

```
StoreVersionState updatedState = GenericData.get().deepCopy(
    storeVersionState.getSchema(), 
    storeVersionState);
updatedState.set("compressionDictionary", ByteBuffer.wrap("ignored".getBytes()));
```

And I think you would not need to loop manually on the fields at all.

Reply:
Oh, sorry, I missed that you are already doing `updatedState = new StoreVersionState();` inside the if below. So then the code is fine, although you might still want to use the built-in deep copier function, if you'd like...

Reply:
yeah, the code is doing a shallow copy and only update the `compressionDictionary` field, which should be more lightweight than a deep copy without modifying the original param.